packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -30,6 +30,57 @@ class SizeAnalyzer {
 
   static const int tableWidth = 80;
 
+  /// Analyze the [aotSnapshot] in an uncompressed output directory.
+  Future<Map<String, dynamic>> analyzeAotSnapshot({
```

Comment:
I'm somewhat surprised this logic isn't extracted from the function below since it more or less does the same thing. Shouldn't the next one essentially just call unzip then call this?

Reply:
yeah good point, I think I could give this another go.

Reply:
Does this mean only Android is putting out a zipped file. All the other platforms are just directories? That's neat.

Reply:
Yeah, while iOS and macOS would eventually end up in a zip archive the tool itself doesn't build this - its done through xcode. The .app is pretty close, modulo some compression, but I think it is really more like an android appbundle in that Apple applies more thinning/compression before downloading.

For windows and linux there is no real "distribution" format yet, so the directory is the next best thing.